src/shared/parameter/cpp_config_test.cpp

```diff
@@ -461,13 +461,39 @@ class TestParameterMutation : public YamlLoadFixture
 TEST_F(TestParameterMutation, DynamicParametersTest)
 {
     // This creates a shared ptr pointing to a ThunderbotsConfig which can be mutated
-    const std::shared_ptr<ThunderbotsConfig> MutableDynamicParameters =
+    const std::shared_ptr<ThunderbotsConfig> mutable_dynamic_parameters =
         std::make_shared<ThunderbotsConfig>();
 
     // This creates an immutable ThunderbotsConfig w/ proper const correctnesss
-    const std::shared_ptr<const ThunderbotsConfig> DynamicParameters =
-        std::const_pointer_cast<const ThunderbotsConfig>(MutableDynamicParameters);
+    const std::shared_ptr<const ThunderbotsConfig> dynamic_params =
+        std::const_pointer_cast<const ThunderbotsConfig>(mutable_dynamic_parameters);
+
+    mutate_all_parameters(mutable_dynamic_parameters);
+    assert_mutation(dynamic_params, config_yaml);
+}
+
+TEST_F(TestParameterMutation, ProtoGenerationTest)
+{
+    // This creates a shared ptr pointing to a ThunderbotsConfig which can be mutated
+    const std::shared_ptr<ThunderbotsConfig> mutable_dynamic_parameters =
+        std::make_shared<ThunderbotsConfig>();
+
+    // This creates an immutable ThunderbotsConfig w/ proper const correctnesss
+    const std::shared_ptr<const ThunderbotsConfig> dynamic_params =
+        std::const_pointer_cast<const ThunderbotsConfig>(mutable_dynamic_parameters);
+
+    // Mutate all the parameters
+    mutate_all_parameters(mutable_dynamic_parameters);
+
+    // Save as a proto
+    auto proto = mutable_dynamic_parameters->toProto();
+
+    // Create a new top level config
+    const std::shared_ptr<ThunderbotsConfig> mutable_dynamic_param_loaded_from_proto =
+        std::make_shared<ThunderbotsConfig>();
 
-    mutate_all_parameters(MutableDynamicParameters);
-    assert_mutation(DynamicParameters, config_yaml);
+    // Load top level config from the proto and make sure that the mutations across all
+    // configs are reflected
+    mutable_dynamic_param_loaded_from_proto->loadFromProto(proto);
+    assert_mutation(mutable_dynamic_param_loaded_from_proto, config_yaml);
```

Comment:
wondering why these functions are snake_case lol

Reply:
not sure lol, `snake_case` is superior but ill change it to camelCase for consistency sake